File: annotation-processor-testing/src/main/java/com/tickaroo/tikxml/annotationprocessing/elementlist/Catalogue.java
/*
 * Copyright (C) 2015 Hannes Dorfmann
 * Copyright (C) 2015 Tickaroo, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 */

package com.tickaroo.tikxml.annotationprocessing.elementlist;

import com.tickaroo.tikxml.annotation.Element;
import com.tickaroo.tikxml.annotation.Path;
import com.tickaroo.tikxml.annotation.Xml;
import java.util.List;

/**
 * @author Hannes Dorfmann
 */
@Xml
public class Catalogue {

  @Path("books")
  @Element
  List<Book> books;

  @Override public boolean equals(Object o) {
    if (this == o) return true;
    if (!(o instanceof Catalogue)) return false;

    Catalogue catalogue = (Catalogue) o;

    return books != null ? books.equals(catalogue.books) : catalogue.books == null;
  }

  @Override public int hashCode() {
    return books != null ? books.hashCode() : 0;
  }
}

File: annotationprocessingtesting-kt/src/main/java/com/tickaroo/tikxml/annotationprocessing/elementlist/Catalogue.kt
/*
 * Copyright (C) 2015 Hannes Dorfmann
 * Copyright (C) 2015 Tickaroo, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 */

package com.tickaroo.tikxml.annotationprocessing.elementlist

import com.tickaroo.tikxml.annotation.Element
import com.tickaroo.tikxml.annotation.Path
import com.tickaroo.tikxml.annotation.Xml

/**
 * @author Hannes Dorfmann
 */
@Xml
class Catalogue {

    @Path("books")
    @Element
    var books: List<Book>? = null

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is Catalogue) return false

        val catalogue = other as Catalogue?

        return if (books != null) books == catalogue!!.books else catalogue!!.books == null
    }

    override fun hashCode(): Int {
        return if (books != null) books!!.hashCode() else 0
    }
}
